### DISCUSSION TITLE
HelperIndex(n) trigger

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#1887] [Engine development]
Today I discovered helperIndex(n) trigger. This trigger theoretically allowing you to get velocity, horizontal and vertical position and other parameters of enemy helper if it exists. That's very good. In vanilla Mugen It was not so easy task, to get coordinates of enemy helper to allow AI to jump over it (on current moment I'm using special helper for AI, that shows coordinates and speed of opponent helper. I'm not very like how it works, especially if several opponent helpers on the screen). In principle, it is theoretically clear how you can determine the helper closest to you, even if your opponent uses several helpers. 

Can we get something like this trigger for projectiles in the future? On current moment I don't know a way to get precise coordinates of opponent projectiles (exactly **projectiles**, not helpers) in Mugen. Method by Inktrebuchet shows theoretical coordinates of projectile based on projectile release time and projectile contact time with you. This is a very rough estimate and it’s no good if the opponent releases several projectiles.

Thank you!

**Comment by Kiwi1444** [comment]
Tested HelperIndex(n) trigger a little bit. Is it normal, that even after destroying opponent helper, trigger continues to show last known position of helper (enemy,HelperIndex(1),pos x is nonzero)?

It's very strange, but after destroying opponent helper, trigger enemy,HelperIndexExist(1) continues to show 1. I guess I don't understand how it works...

**Reply by Kiwi1444** [reply]
Is this trigger behavior normal? Can anyone explain? If it normal, how can I confirm, that helper exists without playeridexist trigger? Something like:
`playeridexist(enemy,HelperIndex(1),ID)`

**Reply by Wftk** [reply]
It should be fixed now.Also check the wiki cuz trigger been refactored.
